pandas/core/internals/managers.py

```diff
@@ -1505,7 +1505,11 @@ def get_slice(self, slobj, axis=0):
         if axis >= self.ndim:
             raise IndexError("Requested axis not found in manager")
 
-        return type(self)(self._block._slice(slobj), self.index[slobj], fastpath=True)
+        return type(self)(
+            self._block._slice(slobj),
+            self.index._getitem_deprecate_nd(slobj, warn=False),
```

Comment:
`Index.__getitem__` shouldn't be issuing a warning when the arg is a slice object, should it?  then is get_slice being called incorrectly?

Reply:
This is what's causing the warning

```python
In [5]: pd.Index(['a', 'b'])[:, None]
/Users/taugspurger/.virtualenvs/pandas-dev/bin/ipython:1: DeprecationWarning: Support for multi-dimensional indexing (e.g. `index[:, None]`) on an Index is deprecated and will be removed in a future version.  Convert to a numpy array before indexing instead.
  #!/Users/taugspurger/Envs/pandas-dev/bin/python
Out[5]:
array([['a'],
       ['b']], dtype=object)

```

Which I think looks correct.

Reply:
What im saying/asking is that BlockManager.get_slice should have `slobj` be a slice object, not a tuple `(some_slice, None)`.  So it looks like we're calling get_slice incorrectly.

If this is a blocker, we can put in the warning now and I can track down the usage in a followup